homeassistant/components/vesync/light.py

```diff
@@ -68,17 +88,105 @@ def turn_on(self, **kwargs):
             brightness = round((brightness / 255) * 100)
             # clamp to 1-100
             brightness = max(1, min(brightness, 100))
-            self.dimmer.set_brightness(brightness)
+            self.device.set_brightness(brightness)
         # Avoid turning device back on if this is just a brightness adjustment
         if not self.is_on:
             self.device.turn_on()
 
     @property
-    def supported_features(self):
-        """Get supported features for this entity."""
-        return SUPPORT_BRIGHTNESS
+    def color_mode(self):
+        """Set color mode for this entity."""
+        return COLOR_MODE_BRIGHTNESS
 
     @property
-    def brightness(self):
-        """Get dimmer brightness."""
-        return round((int(self.dimmer.brightness) / 100) * 255)
+    def supported_color_modes(self):
+        """Flag supported color_modes (in an array format)."""
+        return [COLOR_MODE_BRIGHTNESS]
+
+
+class VeSyncTunableWhiteLightHA(VeSyncBaseLight, LightEntity):
+    """Representation of a VeSync Tunable White Light device."""
+
+    def __init__(self, device):
+        """Initialize the VeSync Tunable White Light device."""
+        super().__init__(device)
+        self.device = device
+
+    def turn_on(self, **kwargs):
+        """Turn the device on."""
+        if ATTR_BRIGHTNESS in kwargs:
+            # get brightness from HA data
+            brightness = int(kwargs[ATTR_BRIGHTNESS])
+            # convert to percent that vesync api expects
+            brightness = round((brightness / 255) * 100)
+            # clamp to 1-100
+            brightness = max(1, min(brightness, 100))
+            # pass value to pyvesync library api
+            self.device.set_brightness(brightness)
+
+        # set white temperature
+        if ATTR_COLOR_TEMP in kwargs:
+            # get white temperature from HA data
+            color_temp = int(kwargs[ATTR_COLOR_TEMP])
+
+            # convert Mireds to Percent value that api expects
+            color_temp = round(
+                ((color_temp - self.min_mireds) / (self.max_mireds - self.min_mireds))
+                * 100
+            )
+
+            # flip cold/warm to what pyvesync api expects
+            color_temp = 100 - color_temp
+
+            # ensure value between 0-100
+            color_temp = max(0, min(color_temp, 100))
+
+            # pass value to pyvesync library api
+            self.device.set_color_temp(color_temp)
+
+        # Avoid turning device back on if this is just a brightness adjustment
+        if not self.is_on:
```

Comment:
Will setting the colortemp or brightness turn on the device.  If so we should check to see if we did some of those and return instead as the device could be flipped on/off outside of home assistant before the next update and then the state is out of sync.

Reply:
I tested the heck out of it, using home assistant and the native vesync app simultaneously, I couldn't came to a conclusion that there is any problem. It just works. the states updates on both sides pretty quick. I don't know if I'm not understanding some more complex innerworks here, and I really don't know what to change here, if it does need a change. for me it's working as is.